### PR TITLE
OperationEditor: Fix rendering of operation editor items

### PR DESCRIPTION
**Pull request by ivanahuckova**
I have introduced an issue when I was refactoring OperationEditor (at that time in `grafana/grafana` , but because I wasn't passing props correctly to component, it was doing a lot of mounting - unmounting https://github.com/grafana/grafana/commit/7be8301a265c7b228d8cee77188a094504fa1335. I will have to do this fix also in `grafana/grafana`

This PR takes previously called `StyledOperationHeader` and moves it to separate file and makes it a proper component that doesn't cause -remounting of each operation editor. 

Fixed:

https://github.com/grafana/grafana-experimental/assets/30407135/168a846f-ceaa-43f5-9236-35f298d6b754

Broken:

https://github.com/grafana/grafana-experimental/assets/30407135/7c5b7724-ab1c-4b59-bde3-3b0769b9676d

How to test:
- If you can use `yarn link` and test this locally, do that.
- For me it works only by:
  -  1. renaming package in `package.json` to `custom`
  - in `grafana/grafana` `package.json` add `"custom": "file: ../grafana-experimental"`
  - import `OperationList` in Loki from `custom` 
  - check the recording and do that/other interactions

### CODE REVIEW

File: src/VisualQueryBuilder/components/OperationEditorBody.tsx [hunk 1]
```diff
@@ -0,0 +1,288 @@
...
+        operation={operation}
+        dragHandleProps={provided.dragHandleProps}
+        definition={definition}
+        index={index}
+        onChange={onChange}
+        onRemove={onRemove}
+        queryModeller={queryModeller}
+      />
+      <div className={styles.body}>
+      {
```
Comment (matyax) [inline src/VisualQueryBuilder/components/OperationEditorBody.tsx:84]: This looks weirdly styled, maybe it needs to pass through prettier.

File: src/VisualQueryBuilder/components/OperationEditorBody.tsx [hunk 2]
```diff
@@ -0,0 +1,288 @@
...
+    callParamChangedThenOnChange(definition, update, index, operation.params.length, onChange);
+  };
+
+  const onRemoveRestParam = (paramIdx: number) => {
+    const update: QueryBuilderOperation = {
+      ...operation,
+      params: [...operation.params.slice(0, paramIdx), ...operation.params.slice(paramIdx + 1)],
+    };
+    callParamChangedThenOnChange(definition, update, index, paramIdx, onChange);
+  };
```
Comment (matyax) [inline src/VisualQueryBuilder/components/OperationEditorBody.tsx:52]: I would conside `useCallback` these, now that we're at it, as it doesn't need to change the children on every render.

File: src/VisualQueryBuilder/components/OperationEditorBody.tsx [hunk 3]
```diff
@@ -0,0 +1,288 @@
...
+
+  // Handle adding button for rest params
+  let restParam: React.ReactNode | undefined;
+  if (definition.params.length > 0) {
+    const lastParamDef = definition.params[definition.params.length - 1];
+    if (lastParamDef.restParam) {
+      restParam = renderAddRestParamButton(lastParamDef, onAddRestParam, index, operation.params.length, styles);
+    }
+  }
```
Comment (matyax) [inline src/VisualQueryBuilder/components/OperationEditorBody.tsx:61]: And `useMemo` this.

**Review by matyax — APPROVED** [review]
LGTM!

File: src/VisualQueryBuilder/components/OperationEditorBody.tsx (hunk 1 above)
Comment (ivanahuckova) [reply src/VisualQueryBuilder/components/OperationEditorBody.tsx]: Fixed in https://github.com/grafana/grafana-experimental/pull/112/commits/eadd757e3bf0f18bc6e6418139927eeb29a1c14a

**Pull request merged**